File: Examples/fsharp/test/MyCdkAppTests.fs
namespace MyCdkApp.Tests

open Xunit
open Amazon.CDK
open Amazon.CDK.Assertions
open MyCdkApp
open System.Collections.Generic

type MyCdkAppTests() =

    [<Fact>]
    member _.``SQS Queue and SNS Topic Created``() =
        let app = App()
        let stack = MyCdkAppStack(app, "TestStack")
        let template = Template.FromStack(stack)

        let queueProps = Dictionary<string, obj>()
        queueProps.Add("VisibilityTimeoutSeconds", box 300)
        template.HasResourceProperties("AWS::SQS::Queue", queueProps)

        let topicProps = Dictionary<string, obj>()
        template.HasResourceProperties("AWS::SNS::Topic", topicProps)

File: Samples/csharp/test/MyCdkAppTests.cs
using Amazon.CDK;
using Amazon.CDK.Assertions;
using Xunit;
using MyCdkApp;

namespace MyCdkApp.Tests
{
    public class MyCdkAppTests
    {
        [Fact]
        public void SQSQueueAndSNSTopicCreated()
        {
            var app = new App();
            var stack = new MyCdkAppStack(app, "TestStack");

            var template = Template.FromStack(stack);

            template.HasResourceProperties("AWS::SQS::Queue", new Dictionary<string, object>
            {
                ["VisibilityTimeoutSeconds"] = 300
            });

            template.HasResourceProperties("AWS::SNS::Topic", new Dictionary<string, object>());
        }
    }
}